Refactor requirements to be cross-platform

AymenQ opened the pull request:
Moved from conda's auto-generated requirements.txt which was OSX
specific to a manually created environment.yml file which is much
simpler and cross platform. Also allows conda and pip requirements to be
defined in the same file. Fixes #16.

Signed-off-by: Aymen Qader <aq1g17@soton.ac.uk>

khanh111 commented:
How did you create this shortlist of dependencies and validate that it encompasses the dependencies of the scripts? Some of the packages listed in the current `requirements.txt` are not present in the `environment.yml` file.

AymenQ commented:
> First of all, jazakallahkhair for taking this task on. I have one quick question before we can land this to ensure we don't have dependency issues moving forward.
> 
> How did you create this shortlist of dependencies and validate that it encompasses the dependencies of the scripts? Some of the packages listed in the current `requirements.txt` are not present in the `environment.yml` file.

Sorry, I should have been more clear in the pull request. I've gone through the modules required by the scripts, and added them to the environment.yml as needed. This gives a much shorter list than the output of `conda list -e` which outputs all the packages installed regardless of whether they were installed as dependencies for other packages.

When you run `conda env create -f environment.yml` and then run `conda list -e`, the list is very similar barring some platform-specific variation in packages.

I did miss off scikit-learn, but I've just added it and checked the imports for all the scripts so it should encompass everything now.
